pkg/util/cluster_test.go

```diff
@@ -0,0 +1,134 @@
+package util
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	karmadaclientset "github.com/karmada-io/karmada/pkg/generated/clientset/versioned"
+	karmadaclientsetfake "github.com/karmada-io/karmada/pkg/generated/clientset/versioned/fake"
+)
+
+func newCluster(name string) *clusterv1alpha1.Cluster {
+	return &clusterv1alpha1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec:   clusterv1alpha1.ClusterSpec{},
+		Status: clusterv1alpha1.ClusterStatus{},
+	}
+}
+
+func withAPIEndPoint(cluster *clusterv1alpha1.Cluster, apiEndPoint string) *clusterv1alpha1.Cluster {
+	cluster.Spec.APIEndpoint = apiEndPoint
+	return cluster
+}
+
+func withSyncMode(cluster *clusterv1alpha1.Cluster, syncMode clusterv1alpha1.ClusterSyncMode) *clusterv1alpha1.Cluster {
+	cluster.Spec.SyncMode = syncMode
+	return cluster
+}
+
+func TestCreateOrUpdateClusterObject(t *testing.T) {
+	fakeClient := karmadaclientsetfake.NewSimpleClientset()
+	type args struct {
+		controlPlaneClient karmadaclientset.Interface
+		clusterObj         *clusterv1alpha1.Cluster
+		mutate             func(*clusterv1alpha1.Cluster)
+		aop                func() func()
```

Comment:
what does `aop` mean?

Reply:
> what does aop mean?

Oh, It's coding habits, learned from others,Do you have better suggestion to replace it

Reply:
I don't know what's the meaning of the `aop` yet, so no suggestion now.

Reply:
> I don't know what's the meaning of the aop yet

function `aop` used for  create dependent resource before run tested logic, such as use fakeClient create cluster object and so on